File: bricks/native_plugin/__brick__/{{name.snakeCase()}}/macos/Classes/{{name.pascalCase()}}Plugin.swift
import Cocoa
import FlutterMacOS

public class {{name.pascalCase()}}Plugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "{{package_prefix.snakeCase()}}_{{name.snakeCase()}}",
            binaryMessenger: registrar.messenger
        )
        let instance = {{name.pascalCase()}}Plugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getData":
            let processInfo = ProcessInfo.processInfo
            let data: [String: Any] = [
                "platform": "macos",
                "operatingSystemVersion": processInfo.operatingSystemVersionString,
                "hostName": processInfo.hostName,
                "processName": processInfo.processName,
                "processorCount": processInfo.processorCount
            ]
            result(data)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: bricks/native_plugin/__brick__/{{name.snakeCase()}}/android/src/main/kotlin/com/{{package_prefix.snakeCase()}}/{{name.snakeCase()}}/{{name.pascalCase()}}Plugin.kt
package com.{{package_prefix.snakeCase()}}.{{name.snakeCase()}}

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** {{name.pascalCase()}}Plugin */
class {{name.pascalCase()}}Plugin: FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "{{package_prefix.snakeCase()}}_{{name.snakeCase()}}")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
        when (call.method) {
            "getData" -> {
                val data = hashMapOf<String, Any>(
                    "platform" to "android",
                    "version" to android.os.Build.VERSION.SDK_INT.toString(),
                    "model" to android.os.Build.MODEL,
                    "manufacturer" to android.os.Build.MANUFACTURER
                )
                result.success(data)
            }
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
